### PR TITLE
feat: reuse the same bigquery table for multiple model versions for Arize BQ Sink

### PR DESCRIPTION
**Pull request by khorshuheng**
# Description

The current sink create a new bigquery table per model version. This makes it harder to implement Arize ground truth ingestion, because the ground truth provided by the users are typically model version agnostic.

# Modifications

- A single table will be used per model id, rather than model version
- A new column, model_version, is added.
- session id and row id are used in favor of prediction id as the concept of prediction id in Arize differs from Merlin

# Tests

# Checklist
- [ ] Added PR label
- [ ] Added unit test, integration, and/or e2e tests
- [ ] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduces API changes

# Release Notes

```release-note

```

### CODE REVIEW

File: python/observation-publisher/publisher/observation_sink.py [hunk 1]
```diff
@@ -207,13 +287,31 @@ def schema_fields(self) -> List[SchemaField]:
 
     @property
     def write_location(self) -> str:
-        table_name = f"prediction_log_{self._model_id}_{self._model_version}".replace(
-            "-", "_"
-        ).replace(".", "_")
-        return f"{self._project}.{self._dataset}.{table_name}"
+        table_name = f"prediction_log_{self._model_id}".replace("-", "_").replace(
```
Comment (tiopramayudi) [inline python/observation-publisher/publisher/observation_sink.py:290]: thinking out loud, should we make it as configuration?

File: python/observation-publisher/publisher/observation_sink.py [hunk 2]
```diff
@@ -207,13 +287,31 @@ def schema_fields(self) -> List[SchemaField]:
...
+            try:
+                response = self._client.insert_rows_from_dataframe(
+                    dataframe=dataframe, table=self._table
+                )
+                errors = [error for error_chunk in response for error in error_chunk]
+                if len(errors) > 0:
+                    print(
+                        f"Errors when inserting rows to BigQuery, retrying attempt {i}/{self.retry.retry_attempts}"
+                    )
+                    time.sleep(self.retry.retry_interval_seconds)
```
Comment (tiopramayudi) [inline python/observation-publisher/publisher/observation_sink.py:306]: will it create build up on the unprocessed kafka messages?

File: python/observation-publisher/publisher/observation_sink.py [hunk 3]
```diff
@@ -207,13 +287,31 @@ def schema_fields(self) -> List[SchemaField]:
...
+                else:
+                    return
+            except NotFound as e:
+                print(
+                    f"Table not found: {e}, retrying attempt {i}/{self.retry.retry_attempts}"
+                )
+                time.sleep(self.retry.retry_interval_seconds)
```
Comment (tiopramayudi) [inline python/observation-publisher/publisher/observation_sink.py:313]: why do we need to retry for not found table?

File: python/observation-publisher/publisher/observation_sink.py (hunk 2 above)
Comment (khorshuheng) [reply python/observation-publisher/publisher/observation_sink.py:306]: Yes, it does, so retry is only useful in the event where the incoming message rate is low enough.

File: python/observation-publisher/publisher/observation_sink.py (hunk 3 above)
Comment (khorshuheng) [reply python/observation-publisher/publisher/observation_sink.py:313]: When the Bigquery table is first created, it does take ~10 seconds for the table to be ready.

**Review by tiopramayudi — APPROVED** [review]
LGTM ! Thanks @khorshuheng

**Pull request merged**